### DISCUSSION TITLE
Yolov5 wrong detection on webcam

### DISCUSSION THREAD

**Discussion by vbsutarsir** [ultralytics/yolov5#10023] [Q&A]
Hello,
I've trained yolov5s for my custom single class cattle dataset and obtained best.pt weight. Tested inference with detect.py for images and youtube urls, got 80 to 90% detection accuracy. But problem is that on source 0(webcam) its showing cattle class for other objects detection (for example person). Please guide!

**Comment by glenn-jocher** [comment]
@vbsutarsir some formats do not include class names as metadata, but PyTorch models should. You can always pass the data.yaml as --data to detect.py if your model is missing class name info, otherwise it may default to COCO class names.

**Reply by vbsutarsir** [reply]
Sir,
Please look following contents of custom_data.yaml.

train: ../train_data/images/train/  # train images 
val: ../train_data/images/val/  # val images 
test:  # test images (optional)

# Classes
nc: 1  # number of classes
names: [ 'Cattle' ]  # class names

**Reply by vbsutarsir** [reply]
Same custom .yaml file was used during the training!

**Reply by vbsutarsir** [reply]
You mean during inference: its necessary to supply custom_data.yaml as parser argument? 
Like--> !python detect.py --weights best.pt --img 640 --source 0 --data/custom_data.yaml
